### PR TITLE
fix(dropdown): enable Dropdown to be cleared and reset

### PR DESCRIPTION
**Pull request by estrattonbailey**
https://user-images.githubusercontent.com/4732330/107270063-14727b00-6a10-11eb-9da4-a82055918a81.mov

### CODE REVIEW

File: src/Dropdown.tsx [hunk 1]
```diff
@@ -265,11 +265,21 @@ export function Dropdown ({
...
-  )
+    )
+  }, [value, items, placeholder])
+
+  const [cta, ctaSet] = React.useState(computedCTA)
+
+  // will only call ctaSet if computedCTA changes
+  React.useEffect(() => {
+    ctaSet(computedCTA)
+  }, [computedCTA])
```
Comment (angelac17) [inline src/Dropdown.tsx:282]: Wrapping my head around this part. Just read about `useMemo` to learn about it. Is there a reason we need to have computedCTA in a `useMemo` hook vs listening to the same list of dependencies (value, items, placeholder) in the `useEffect` call?

**Review by angelac17 — APPROVED** [review]
Amazing, thank you so much for updating this Dropdown! 💯

File: src/Dropdown.tsx (hunk 1 above)
Comment (estrattonbailey) [reply src/Dropdown.tsx:282]: That's a great question. Now that I think about it, we could probably do this another way that's slightly more simple...

My initial thinking was just that we need this value to populate cta state synchronously, and I didn't want to duplicate logic. But you're right, after the initial render it's only used in one location, so we could break this down further.

Will push up a thought and we can discuss that!

File: src/Dropdown.tsx [hunk 2]
```diff
@@ -265,11 +281,12 @@ export function Dropdown ({
...
-    get(items.filter(i => i.value === value)[0], 'label') ||
-      placeholder ||
-      'Please select'
-  )
+  const [cta, ctaSet] = React.useState(getCTA({ items, value, placeholder }))
+
+  // will only call ctaSet if computedCTA changes
+  React.useEffect(() => {
+    ctaSet(getCTA({ items, value, placeholder }))
+  }, [items, value, placeholder])
```
Comment (estrattonbailey) [inline src/Dropdown.tsx:289]: @angelac17 broke this down further: 
- synchronously sets initial state
- updates state if pertinent values change

Comment (angelac17) [reply src/Dropdown.tsx:289]: This look great! Thanks for putting out an update so quickly🥇

**Review by angelac17 — APPROVED** [review]
This is such a nice update to our Dropdown, thanks for working on it! :)

**Pull request merged**